### PR TITLE
[RedHat-8] Populate validate with all the controls for aws_parallelcluster_install

### PR DESCRIPTION
**Pull request by francesco-giordano**
### Description of changes
* [Add search-spec-validation.sh to show diff between kitchen.validate.yml](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/da1cc4ccb9c55aad038b9bdccddf1a4a437168f3)
  * Utility script to print the difference between the controls in kitchen.validate.yml and in the directories of the inspec tests 
* [Mov efa_system_settings_configured control in config directory](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/c8d80cce6edffc676be05baa5a17153bcc76aec2)
  * Minor move of an existing tests from install to config inspec test directory to reflect the usage
* [Add controls to kitchen.validate.yml and removed unused configuration](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/1511bee4bd1fe2e05962fe6eb7f2dd62dad4b870)
  * Added the missing inspec test and moved attribute at "lower level" in order to keep the kitchen files as general as possible
* [Add validate_os_type for redhat8](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/52b2487199c601276851b8abe0a717c5f2fd0a87)
  * Add a missing validator on the OS version for RedHat8
* [Fix chrony recipe and spec in case of redhat_ubi](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/97960567258aacb3759639c1bb64696a5d42d236)
  * Fix a bug in the chrony recipe and spec that was not able to deal with the docker image of RedHat8
* [Move cluster attribute from global kitchen config to clusterstatusmgtd_init_slurm test](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/4f2983cdf1906186f59c1be3fe5e9c49bdbdc7b2)
  * Moved attribute at "lower level" in order to keep the kitchen files as general as possible
* [Add to lustre repo skip_if_unavail](https://github.com/aws/aws-parallelcluster-cookbook/pull/1791/commits/69261417e7c09327b681911c4b218d4e1656096e)

### Tests
* Locally tested on docker

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-install/resources/package_repos/package_repos.rb [hunk 1]
```diff
@@ -29,11 +29,7 @@
     # the epel recipe doesn't work on aarch64, needs epel-release package
     package 'epel-release' if node['platform_version'].to_i == 7 && node['kernel']['machine'] == 'aarch64'
 
-    unless node['platform_version'].to_i < 7
-      execute 'yum-config-manager_skip_if_unavail' do
-        command "yum-config-manager --setopt=\*.skip_if_unavailable=1 --save"
-      end
-    end
+    action_centos_skip_if_unavail
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-install/resources/package_repos/package_repos.rb:32]: Do we still need it here?

File: kitchen.global.yml [hunk 2]
```diff
@@ -3,9 +3,6 @@ provisioner:
   attributes:
     cluster:
       # right now tests depend on this parameter: we will try to remove the dependency later
-      base_os: alinux
+      base_os: alinux2
```
Comment (awslitvit) [inline kitchen.global.yml]: Shouldn't we remove it?

File: kitchen.global.yml [hunk 3]
```diff
@@ -3,9 +3,6 @@ provisioner:
...
       region: us-east-1
-      dcv:
-        server: {}
-      parallelcluster-node-version: '3.4.1'
-      node_type: 'HeadNode'
+      parallelcluster-node-version: '3.4.1'
```
Comment (awslitvit) [inline kitchen.global.yml:7]: Do we still need this?

File: kitchen.validate.yml [hunk 4]
```diff
@@ -53,7 +42,19 @@ suites:
...
         - openssh_installed
         - ami_cleanup_file_created
-        # TODO: enable following controls when testing install.rb recipe
-        # - efa_conflicting_packages_removed
-        # - efa_prereq_packages_installed
-        # - efa_installed
+        - c_states_kernel_configuration
+        - chrony_installed_and_configured
+        - clusterstatusmgtd_files_created
+        - disable_selinux
```
Comment (awslitvit) [inline kitchen.validate.yml]: Nitpick. Could we rename it to `selinux_disabled`?

File: kitchen.validate.yml [hunk 5]
```diff
@@ -53,7 +42,19 @@ suites:
         - cron_disabled_selected_daily_and_weekly_jobs
         - openssh_installed
         - ami_cleanup_file_created
-        # TODO: enable following controls when testing install.rb recipe
-        # - efa_conflicting_packages_removed
-        # - efa_prereq_packages_installed
-        # - efa_installed
+        - c_states_kernel_configuration
```
Comment (awslitvit) [inline kitchen.validate.yml]: Nitpick. Could we rename it to past participle?

File: kitchen.validate.yml [hunk 6]
```diff
@@ -53,7 +42,19 @@ suites:
...
+        - disable_selinux
+        - efa_conflicting_packages_removed
+        - efa_installed
+        - efa_prereq_packages_installed
+        - efs_utils_installed
+        - gc_thresh_values_configured
+        - license_readme_created
+        - mysql_client_installed
+        - mysql_client_source_node_created
+        - node_attributes
```
Comment (awslitvit) [inline kitchen.validate.yml]: Nitpick. Could we rename it to past participle?

File: libraries/helpers.rb [hunk 7]
```diff
@@ -190,6 +190,9 @@ def validate_os_type
   when 'centos'
     current_os = "centos#{node['platform_version'].to_i}"
     raise_os_not_match(current_os, node['cluster']['base_os']) if node['cluster']['base_os'] != current_os
+  when 'redhat'
```
Comment (awslitvit) [inline libraries/helpers.rb:193]: out-of-scope for this PR but we need to move this code to a resource as it is OS-dependent.

File: test/resources/controls/aws_parallelcluster_config/efa_system_settings_configured_spec.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+control 'efa_system_settings_configured' do
```
Comment (awslitvit) [inline test/resources/controls/aws_parallelcluster_config/efa_system_settings_configured_spec.rb]: Nitpick. It should have `debian` in the control name.

Comment (awslitvit) [inline test/resources/controls/aws_parallelcluster_config/efa_system_settings_configured_spec.rb]: Nitpick. Can we name the file `efa_spec.rb` to make it more generic?

File: cookbooks/aws-parallelcluster-install/resources/package_repos/package_repos.rb (hunk 1 above)
Comment (francesco-giordano) [reply cookbooks/aws-parallelcluster-install/resources/package_repos/package_repos.rb:32]: I think that the goal behind this is to not block the installation of packages if some repository is temporarily offline.

File: kitchen.global.yml (hunk 2 above)
Comment (francesco-giordano) [reply kitchen.global.yml:6]: In another PR, I'm going to move most of the configuration from `kitchen.global.yml` into other kitchen file.

Comment (francesco-giordano) [reply kitchen.global.yml]: Done

File: kitchen.global.yml (hunk 3 above)
Comment (francesco-giordano) [reply kitchen.global.yml:7]: This attribute is needed to be able to install an already published parallelcluster-node-version

File: kitchen.validate.yml (hunk 4 above)
Comment (francesco-giordano) [reply kitchen.validate.yml]: Done

File: kitchen.validate.yml (hunk 5 above)
Comment (francesco-giordano) [reply kitchen.validate.yml]: Done

File: kitchen.validate.yml (hunk 6 above)
Comment (francesco-giordano) [reply kitchen.validate.yml]: Done

File: test/resources/controls/aws_parallelcluster_config/efa_system_settings_configured_spec.rb (hunk 8 above)
Comment (francesco-giordano) [reply test/resources/controls/aws_parallelcluster_config/efa_system_settings_configured_spec.rb]: Done

Comment (francesco-giordano) [reply test/resources/controls/aws_parallelcluster_config/efa_system_settings_configured_spec.rb]: Done

File: libraries/helpers.rb (hunk 7 above)
Comment (francesco-giordano) [reply libraries/helpers.rb:193]: Agreed, adding to the task.

File: cookbooks/aws-parallelcluster-install/resources/package_repos/package_repos.rb (hunk 1 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-install/resources/package_repos/package_repos.rb:32]: Why only on CentOS7 though?

File: kitchen.global.yml (hunk 3 above)
Comment (awslitvit) [reply kitchen.global.yml:7]: Do we really want to install `3.4.1`?

Comment (francesco-giordano) [reply kitchen.global.yml:7]: It is ok since it is not used nor checked. Any released version is ok.

**Review by awslitvit — APPROVED** [review]

**Pull request merged**